src/azure-cli/azure/cli/command_modules/vm/generated/_params.py

```diff
@@ -0,0 +1,61 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    resource_group_name_type,
+    get_location_type
+)
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('vm ssh-public-key list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+
+    with self.argument_context('vm ssh-public-key show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('ssh_public_key_name', options_list=['--name', '-n', '--ssh-public-key-name'], type=str, help='The '
+                   'name of the SSH public key.', id_part='name')
+
+    with self.argument_context('vm ssh-public-key create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('ssh_public_key_name', options_list=['--name', '-n', '--ssh-public-key-name'], type=str, help='The '
+                   'name of the SSH public key.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), required=False,
+                   validator=get_default_location_from_resource_group)
+        c.argument('tags', tags_type)
+        c.argument('public_key', type=str, help='SSH public key used to authenticate to a virtual machine through ssh. '
```

Comment:
public_key should not be required right?  `az sshkey create --name myKey` should behave
- if user pass in public_key, if key resource exists, updating. otherwise error out
- if user didn't pass public_key,  create key resource, and call generateKeyPair.
Is this correct?

Reply:
`az sshkey create` is used to create. Do not use it to update.